os/kernel/mqueue/mq_send.c

```diff
@@ -154,6 +154,11 @@ int mq_send(mqd_t mqdes, FAR const char *msg, size_t msglen, int prio)
 		return ERROR;
 	}
 
+	if (mq_check(mqdes) != OK) {
+		leave_cancellation_point();
+		return ERROR;
+	}
+
```

Comment:
We need to check the mqdes in all mq functions like send, receive, getattr, setattr, notify, etc..